src/meta/processors/indexMan/CreateTagIndexProcessor.cpp

```diff
@@ -43,6 +43,38 @@ void CreateTagIndexProcessor::process(const cpp2::CreateTagIndexReq& req) {
         return;
     }
 
+    auto prefix = MetaServiceUtils::indexPrefix(space);
+    std::unique_ptr<kvstore::KVIterator> checkIter;
+    auto checkRet = kvstore_->prefix(kDefaultSpaceId, kDefaultPartId, prefix, &checkIter);
+    if (checkRet != kvstore::ResultCode::SUCCEEDED) {
+        resp_.set_code(to(checkRet));
+        onFinished();
+        return;
+    }
+
+    while (checkIter->valid()) {
+        auto val = checkIter->val();
+        auto item = MetaServiceUtils::parseIndex(val);
+        if (item.get_schema_id().getType() != nebula::cpp2::SchemaID::Type::tag_id ||
+            fieldNames.size() > item.get_fields().size()) {
+            continue;
+        }
+
+        for (size_t i = 0; i < fieldNames.size(); i++) {
+            if (fieldNames[i] != item.get_fields()[i].get_name()) {
+                break;
+            }
+
+            if (i == fieldNames.size() - 1) {
+                LOG(ERROR) << "Tag Index " << indexName << " have existed";
+                resp_.set_code(cpp2::ErrorCode::E_EXISTED);
+                onFinished();
+                return;
+            }
+        }
+        checkIter->next();
+    }
```

Comment:
Could we unify the code block? I think the tag and edge will be able to share.

Reply:
Seems you should skip edge or tag by item.get_schema_id().getType()

Reply:
> Seems you should skip edge or tag by item.get_schema_id().getType()

There seems is a logic error, this check you should use tagId instead of id_type. 
you could add a test case : 
create tag1(col1, col2);
create tag2(col1, col2);
create index1 on tag1(col1,col2);
create index2 on tag2(col1, col2) --> here expect success . actual return an error E_EXISTED.

Reply:
This compare just for the same schema